infrastructure/bicep/dfx-api.bicep

```diff
@@ -736,121 +692,19 @@ module nodes 'defi-node.bicep' = [for node in nodeProps: {
-  name: btcPipName
-  location: location
-  sku: {
-    name: 'Standard'
-  }
-  properties: {
-    publicIPAllocationMethod: 'Static'
-    dnsSettings: {
-      domainNameLabel: btcVmName
-    }
-  }
-}
-
-resource btcNsg 'Microsoft.Network/networkSecurityGroups@2020-11-01' = {
-  name: btcNsgName
-  location: location
-  properties: {
-    securityRules: [
-      {
-        name: 'SSH'
-        properties: {
-          protocol: 'TCP'
-          sourcePortRange: '*'
-          destinationPortRange: '22'
-          sourceAddressPrefix: allowedIpRange
-          destinationAddressPrefix: '*'
-          access: 'Allow'
-          priority: 300
-          direction: 'Inbound'
-        }
-      }
-    ]
-  }
-}
-
-resource rpcRule 'Microsoft.Network/networkSecurityGroups/securityRules@2020-11-01' = if (nodeAllowAllIps) {
-  parent: btcNsg
-  name: 'RPC'
-  properties: {
-    protocol: 'TCP'
-    sourcePortRange: '*'
-    destinationPortRange: '8332'
-    sourceAddressPrefix: allowedIpRange
-    destinationAddressPrefix: '*'
-    access: 'Allow'
-    priority: 350
-    direction: 'Inbound'
-  }
-}
-
-resource btcNic 'Microsoft.Network/networkInterfaces@2020-11-01' = {
-  name: btcNicName
-  location: location
-  properties: {
-    ipConfigurations: [
-      {
-        name: 'ipconfig1'
-        properties: {
-          privateIPAllocationMethod: 'Dynamic'
-          publicIPAddress: {
-            id: btcPip.id
-          }
-          subnet: {
-            id: virtualNet.properties.subnets[1].id
-          }
-        }
-      }
-    ]
-    networkSecurityGroup: {
-      id: btcNsg.id
-    }
-  }
-}
-
-resource btcVm 'Microsoft.Compute/virtualMachines@2022-03-01' = {
-  name: btcVmName
-  location: location
-  properties: {
-    hardwareProfile: {
-      vmSize: 'Standard_B2s'
-    }
-    storageProfile: {
-      imageReference: {
-        publisher: 'canonical'
-        offer: '0001-com-ubuntu-server-focal'
-        sku: '20_04-lts-gen2'
-        version: 'latest'
-      }
-      osDisk: {
-        name: btcVmDiskName
-        createOption: 'FromImage'
-        caching: 'ReadWrite'
-        managedDisk: {
-          storageAccountType: 'StandardSSD_LRS'
-        }
-        diskSizeGB: 1023
-      }
-    }
-    osProfile: {
-      computerName: btcVmName
-      adminUsername: btcVmUser
-      adminPassword: btcVmPassword
-    }
-    networkProfile: {
-      networkInterfaces: [
-        {
-          id: btcNic.id
-        }
-      ]
-    }
-    diagnosticsProfile: {
-      bootDiagnostics: {
-        enabled: true
-      }
-    }
+module btcNodes 'btc-node.bicep' = [for node in btcNodeProps: {
+  name: node.name
+  params: {
+    location: location
+    pipName: node.pipName
+    vmName: node.vmName
+    vmDiskName: node.vmDiskName
+    nsgName: node.nsgName
+    nicName: node.nicName
+    vmUser: btcVmUser
+    vmPassword: btcVmPassword
+    subnetId: virtualNet.properties.subnets[1].id
```

Comment:
Just curious 
how many subnets do we have here?
Why not having everything within one subnet (index 0)?

Reply:
Good question ... I had to dig a little deeper 😉  I wanted the VM not to have access to other resources (DB, DeFiChain Nodes), therefore the second subnet.
But we could probably remove one NSG, as they are the same for both VMs.